### PR TITLE
GSoC2024: TestSuite - os API

### PR DESCRIPTION
**Pull request by Sadaf-A**
## Description
This PR adds test cases to the OS API for better coverage.

## Changes proposed
 -  execCommand()
 - [x] executes command in the background
 - [x] executes a command in a specific directory
 - spawnProcess()
 - [x] handles long-running processes
 - getSpawnedProcess()
 - [x] returns empty array when no processes are running
 - [x] returns multiple spawned processes
 - [x] ensures processes have valid ids and pids
 - updateSpawnedProcess()
 - [x] captures stdErr output
 - [x] handles invalid commands gracefully
 - [x] exits with code zero on success
 - [x] handles multiple stdIn commands
 - [x] handles stdInEnd without stdIn
 - getEnv()
 - [x] handles case-sensitive environment variables
 - [x] retrieves value with leading or trailing spaces
 - [x] retrieves value with special characters
 - getEnvs
 - [x] contains HOME or USER environment variable
 - [x] checks case sensitivity of environment variable keys
 - showNotification()
 - [x] displays notification with longer content
 - setTray()
 - [x] works when icon path is missing
 - [x] works with empty menu items array
 - [x] sets a disabled and checked menu item correctly
 - [x] works with a separator in the menu items
 - open()
 - [x] opens a valid URL
 - [x] throws an error for non-string input
 - getPath()
 - [x] returns a valid path
 - [x] throws an error for non-existent directories

## How to test it

 - Run `cd spec && npm test os`

## Next steps

None.

## Deploy notes

None.

### CODE REVIEW

File: spec/os.spec.js [hunk 1]
```diff
@@ -182,6 +369,28 @@ describe('os.spec: os namespace tests', () => {
             assert.ok(typeof envs == 'object');
             assert.ok(envs['PATH'] || envs['Path']);
         });
+
+        it('contains HOME or USER environment variable', async () => {
```
Comment (shalithasuranga) [inline spec/os.spec.js:373]: These are POSIX environment variables, so won't work on Windows. Please conditionally check using `NL_OS`. Thanks

**Review by shalithasuranga — CHANGES_REQUESTED** [review]
Hello @Sadaf-A. Thanks so much for sending this pull request. I posted a few notes on code changes -- we can merge this once we resolve those requested changes :tada:

Edit: please check failing test cases too.. Thanks

**Comment by Sadaf-A** [comment]
> Hello @Sadaf-A. Thanks so much for sending this pull request. I posted a few notes on code changes -- we can merge this once we resolve those requested changes :tada:
> 
> Edit: please check failing test cases too.. Thanks

Hey, Thanks for the review I'll resolve the comments by the EOD

File: spec/os.spec.js (hunk 1 above)
Comment (Sadaf-A) [reply spec/os.spec.js]: Hey, using `NL_OS` throws an error so I used `rpm_command` for now as I think that will be common to all OS. None of the Neutralino global variables are being loaded when I print the variables.

**Comment by shalithasuranga** [comment]
Hello @Sadaf-A, thanks for sending these updates. Could you please check why GNU/Linux and macOS tests are not ending. We can merge this once we get them running :tada:

**Comment by Sadaf-A** [comment]
> Hello @Sadaf-A, thanks for sending these updates. Could you please check why GNU/Linux and macOS tests are not ending. We can merge this once we get them running 🎉

Hey, I think the test cases are stuck because of `open()` trying to open a browser.

**Review by shalithasuranga — APPROVED** [review]
Thanks for checking, could you please create an issue about the `os.open()` test failure to investigate it later. We can merge this now :tada:

**Comment by Sadaf-A** [comment]
> Thanks for checking, could you please create an issue about the `os.open()` test failure to investigate it later. We can merge this now 🎉

Thanks, I think it is also worth noting that Darwin builds are not running as it not able to find a runner to run the test cases.

**Comment by shalithasuranga** [comment]
Fixed it via https://github.com/neutralinojs/neutralinojs/commit/5ee71c002a6ae5433503d6b1de281d7bf76f1b46 thanks for letting me know about it @Sadaf-A  :tada;

**Pull request merged**